### PR TITLE
ENH: override sf for rdist distribution

### PR DESCRIPTION
**Pull request by OmarManzoor**
#### Reference issue

Towards gh-17832

#### What does this implement/fix?

- Override and add the sf method for rdist distribution
- Add some tests

#### Additional information

CC: @mdhaber Could you kindly have a look to see if this change makes sense? If it looks feasible then we can add the reference distribution to set up the tests.

### CODE REVIEW

**Comment by mdhaber** [comment]
It makes sense. Go ahead and revise the tests, and we can probably merge. (Would you also show the plot for arguments `1-x` where `x = np.logspace(-16, -0.5)`, and show the curve for the `mpmath` implementation, too? That would show the important region more clearly.)

**Comment by OmarManzoor** [comment]
@mdhaber 

Here is the plot for
<Details>

```python
c = 541.0
x = np.logspace(-15, -0.5)
x = 1 - x
plt.loglog(x, stats.rdist.sf(x, c), label="rdist sf main", ls="dashed")
plt.loglog(x, rdist_sf(x, c), label="rdist sf pr", ls="dashdot")
plt.legend()
plt.show()
```

</Details>

image

**Comment by mdhaber** [comment]
This is what i was going for.

```python3
from scipy import stats
import numpy as np
from time import perf_counter
import matplotlib.pyplot as plt
rng = np.random.default_rng()
from mpmath import mp
mp.dps = 1000

def rdist_sf_mpmath(x, c):
    x = mp.mpf(x)
    c = mp.mpf(c)
    return float(mp.one - mp.betainc(c/2, c/2, 0, (x+1)/2, regularized=True))

def rdist_sf(x, c):
    return stats.beta._sf((x+1)/2, c/2, c/2)

c = 10
x = np.logspace(-15, -0.5)
mpmath_values = np.array([rdist_sf_mpmath(1-_x, c) for _x in x], dtype=np.float64)
plt.loglog(x, stats.rdist.sf(1-x, c), label="rdist sf main", ls="dashed")
plt.loglog(x, rdist_sf(1-x, c), label="rdist sf pr", ls="dashdot")
plt.loglog(x, mpmath_values, label="rdist mpmath", ls="dotted")
plt.legend()
plt.show()
```
image

This shows that for moderate values of `c`, there is a meaningful improvement because the default implementation, `sf = 1 - cdf`, turns values of the survival function less than ~`1e-16` into zero.

**Comment by OmarManzoor** [comment]
> This is what i was going for.
> 
> ```python
> from scipy import stats
> import numpy as np
> from time import perf_counter
> import matplotlib.pyplot as plt
> rng = np.random.default_rng()
> from mpmath import mp
> mp.dps = 1000
> 
> def rdist_sf_mpmath(x, c):
>     x = mp.mpf(x)
>     c = mp.mpf(c)
>     return float(mp.one - mp.betainc(c/2, c/2, 0, (x+1)/2, regularized=True))
> 
> def rdist_sf(x, c):
>     return stats.beta._sf((x+1)/2, c/2, c/2)
> 
> c = 10
> x = np.logspace(-15, -0.5)
> mpmath_values = np.array([rdist_sf_mpmath(1-_x, c) for _x in x], dtype=np.float64)
> plt.loglog(x, stats.rdist.sf(1-x, c), label="rdist sf main", ls="dashed")
> plt.loglog(x, rdist_sf(1-x, c), label="rdist sf pr", ls="dashdot")
> plt.loglog(x, mpmath_values, label="rdist mpmath", ls="dotted")
> plt.legend()
> plt.show()
> ```
> 
> image
> 
> This shows that for moderate values of `c`, there is a meaningful improvement because the default implementation, `sf = 1 - cdf`, turns values of the survival function less than ~`1e-16` into zero, whereas the implementation that computes the SF directly does not.

I understand. Thank you for the explanation. Should I remove the tests for this?

File: scipy/stats/tests/test_distributions.py [hunk 1]
```diff
@@ -6632,6 +6632,25 @@ def test_rdist_beta(self):
         assert_almost_equal(0.5*stats.beta(c/2, c/2).pdf((x + 1)/2),
                             stats.rdist(c).pdf(x))
 
+    # reference values were computed via mpmath
+    # from mpmath import mp
+    # mp.dps = 200
+    # def rdist_sf_mpmath(x, c):
+    #     x = mp.mpf(x)
+    #     c = mp.mpf(c)
+    #     return float(mp.one - mp.betainc(c/2, c/2, 0, (x+1)/2, regularized=True))
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:6641]:
```suggestion
    #     return float(mp.betainc(c/2, c/2, (x+1)/2, mp.one, regularized=True))
```

**Comment by mdhaber** [comment]
I went ahead and checked the tests as they are. I think the first two would pass in main, so they aren't really needed, but having the second two can't hurt, even if they are mostly a test of the accuracy of `beta.sf`. I refactored
```python3
float(mp.one - mp.betainc(c/2, c/2, (x+1)/2, mp.one, regularized=True))
```
to
```python3
float(mp.betainc(c/2, c/2, (x+1)/2, mp.one, regularized=True))
```
because we might as well calculate the SF directly if we can.
In the future, please use `ReferenceDistribution`, but I'll go ahead and merge as-is now.

**Pull request merged**